plugin.video.cook/resources/lib/scraper.py

```diff
@@ -0,0 +1,70 @@
+# -*- coding: utf-8 -*-
+# Cooking Channel Kodi Video Addon
+#
+from t1mlib import t1mAddon
+import re
+import xbmc
+import xbmcplugin
+import xbmcgui
+import sys
+import html.parser
+import requests
+
+UNESCAPE = html.parser.HTMLParser().unescape
+
+
+class myAddon(t1mAddon):
+
+  def getAddonMenu(self,url,ilist):
+      html = requests.get('https://www.cookingchanneltv.com/videos/players/full-episodes-player', headers=self.defaultHeaders).text
+      a = re.compile('<div class="m-MediaBlock o-Capsule__m-MediaBlock m-MediaBlock--playlist">.+?href="(.+?)".+?data-src="(.+?)".+?HeadlineText">(.+?)<.+?</div',re.DOTALL).findall(html)
+      for (url, thumb, name) in a:
+          name=UNESCAPE(name.strip())
+          thumb = ''.join(['https:',thumb])
+          fanart  = thumb.split('.rend.',1)[0]
+          infoList = {'mediatype':'tvshow',
+                      'TVShowTitle': name,
+                      'Title': name,
+                      'Studio': 'Cooking Channel',
+                      'Plot': name}
+          ilist = self.addMenuItem(name,'GE', ilist, url, thumb, fanart, infoList, isFolder=True)
+      return(ilist)
+
+
+  def getAddonEpisodes(self,url,ilist):
+      url = ''.join(['https:',url])
+      html = requests.get(url, headers=self.defaultHeaders).text
+      vids  = re.compile('"videos"\: \[(.+?)\]',re.DOTALL).search(html).group(1)
+      vids = ''.join(['[',vids,']'])
+      vids = eval(vids)
+      for c in vids:
+          b = dict(c)
+          url = b['releaseUrl']
+          name = UNESCAPE(b['title'])
+          thumb = b['thumbnailUrl']
+          thumb = ''.join(['https://www.cookingchanneltv.com',thumb])
+          fanart = thumb
+          infoList = {'mediatype':'episode',
+                      'Duration': b.get('length'),
+                      'Title': UNESCAPE(b['title']),
+                      'Studio': b.get('publisherId'),
+                      'Plot': UNESCAPE(b.get("description")),
+                      'TVShowTitle': b.get('showTitle')}
+          ilist = self.addMenuItem(name,'GV', ilist, url, thumb, fanart, infoList, isFolder=False)
+      return(ilist)
+
+
+  def getAddonVideo(self,url):
+     html = requests.get(url, headers=self.defaultHeaders).text
+     subs = re.compile('<textstream src="(.+?\.srt)"',re.DOTALL).search(html)
+     if subs != None:
+         subs = subs.group(1)
+     url = re.compile('<video src="(.+?)"',re.DOTALL).search(html).group(1)
+     if url is None:
+         url, msg = re.compile('<ref src="(.+?)".+?abstract="(.+?)"',re.DOTALL).search(html).groups()
+         xbmc.executebuiltin('XBMC.Notification("%s", "%s", %s)' % ( self.addonName, msg , 5000) )
```

Comment:
same issue. Please use always the python API whenever there are no alternatives (xbmc.Dialog().notification()). Also keep in mind XBMC. prefixes don't work anymore in matrix (this would be only `Notification()` if you were to follow the built-in route.

Reply:
Done

Reply:
Done